### PR TITLE
Adding documentation with a srs file

### PR DESCRIPTION
**Pull request by ljte**

### CODE REVIEW

**Review by inauros — CHANGES_REQUESTED** [review]
**README.md**
Please add the following sections:
1. application functions (for example: adding / updating / removing departments, displaying a list of employees, etc.)
2. installation (in the future, this section will be filled with instructions for installing / configuring / running the application locally)

File: documentation/Specification.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+Application should provide:
+
+-   Storing departments and employees in database;
+-   Displaying list of departments;
+-   Displaying list of employees;
+-   Updating the list of departments (adding, editing, removing);
+-   Updating the list of employees (adding, editing, removing);
+-   Possibility to search for employees based on their birthday or if their were born in a specific period;
+-   Filtering departments by the average salary;
```
Comment (inauros) [inline documentation/Specification.md:15]: Please add the following functions:
1. search by user name
2. search by department name
3. filter users by department name

File: documentation/Specification.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+The mode is designed to view the list of departments.
+
+***Main scenario:***
+
+-   User selects item “Departments”;
+-   Application displays the list of departments;
+
+![Image of the departments page](./mockup_imgs/depart_page.png)
+
+Pic 1.1 view of the Department list
```
Comment (inauros) [inline documentation/Specification.md:33]: Please replace "New department: Add" with just one button "New Department".

File: documentation/Specification.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+-   Department name – human-understandable unique department name
+-   Number of employees – the number of employees in the department
+-   Average salary – the average salary of the departments
+
+Aggregate function: Average salary = SUM(Emloyees.employee\_salary) / Number of employees
+
+Filtering by the average salary:
+
+-   User chooses a comparison operator (&lt;, &gt;, =, &lt;=, &gt;=), enters the average salary and the app filter the table;
+
```
Comment (inauros) [inline documentation/Specification.md:50]: Please add an ability to search by department name

File: documentation/Specification.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+### **1.2 Add Department**
+
+***Main scenario:***
+
+-   User clicks the “Add” button on the Departments page;
+-   Application displays the form for adding department;
+-   User enters department data and clicks “Save” button;
+-   If any data is entered incorrectly, incorrect data message is displayed;
```
Comment (inauros) [inline documentation/Specification.md:58]: Please add a mockup for this use case

File: documentation/Specification.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+-   If entered data is valid, then the record is added to the database;
+-   If an error occurs, then the error message is displayed;
```
Comment (inauros) [inline documentation/Specification.md:60]: Please add a mockup for this use case

File: documentation/Specification.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+-   Department’s name – the name of the departments;
+
+### **1.3 Edit Department**
+
+***Main scenario:***
+
+-   User clicks the “Edit” button on the Departments page;
+-   Application displays form with current Department’s data;
+-   User edits the department’s data and clicks “Save”;
+-   If any data is entered incorrectly, the incorrect data message is displayed;
```
Comment (inauros) [inline documentation/Specification.md:86]: Please add a mockup for this use case

File: documentation/Specification.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+-   If entered data is valid, then the change is saved to the database;
+-   If error occurs, the error message is displayed;
```
Comment (inauros) [inline documentation/Specification.md:88]: Please add a mockup for this use case

File: documentation/Specification.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+-   Department’s name – the name of the department;
+
+### **1.4 Removing a Department**
+
+***Main scenario:***
+
+-   User clicks the “Remove” button in the corresponding line;
+-   Then the prompt appears asking the use to confirm removal;
+-   The user confirms the removal;
+-   The recorded is deleted from the database;
```
Comment (inauros) [inline documentation/Specification.md]: Please replace "recorded" with "record" :)

File: documentation/Specification.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+-   If error occurs, the error message is displayed;
```
Comment (inauros) [inline documentation/Specification.md:115]: Please add a mockup for this use case

File: documentation/Specification.md [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+![Image of the delete department prompt](./mockup_imgs/delete_prompt.png)
+
+Pic 1.4 Remove department
+
+***Cancel operation scenario:***
+
+-   User clicks the “Remove” button in the corresponding line;
+-   Then the prompt appears asking the use to confirm removal;
+-   The user cancels the removal;
+-   The Departments page is displayed without any changes;
```
Comment (inauros) [inline documentation/Specification.md:129]: Please add scenarios and mockups for the Employees page.

File: documentation/Specification.md [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+
+***Main scenario:***
+
+-   User clicks the “Remove” button in the corresponding line;
+-   Then the prompt appears asking the use to confirm removal;
+-   The user confirms the removal;
+-   The recorded is deleted from the database;
+-   If error occurs, the error message is displayed;
+-   If record removal went successfully, then the Departments page is displayed without the deleted record;
+
```
Comment (inauros) [inline documentation/Specification.md:118]: What is the expected behavior if a user tries to delete a department when that department has at least one employee?

**Review by inauros — CHANGES_REQUESTED** [review]

File: documentation/Specification.md (hunk 10 above)
Comment (ljte) [reply documentation/Specification.md:129]: Done

File: documentation/Specification.md (hunk 1 above)
Comment (ljte) [reply documentation/Specification.md:15]: Done

File: documentation/Specification.md (hunk 2 above)
Comment (ljte) [reply documentation/Specification.md:33]: Done

File: documentation/Specification.md (hunk 8 above)
Comment (ljte) [reply documentation/Specification.md]: Done

File: documentation/Specification.md (hunk 3 above)
Comment (ljte) [reply documentation/Specification.md:50]: Done

File: documentation/Specification.md (hunk 4 above)
Comment (ljte) [reply documentation/Specification.md:58]: Done

File: documentation/Specification.md (hunk 5 above)
Comment (ljte) [reply documentation/Specification.md:60]: Done

File: documentation/Specification.md (hunk 6 above)
Comment (ljte) [reply documentation/Specification.md:86]: Done

File: documentation/Specification.md (hunk 7 above)
Comment (ljte) [reply documentation/Specification.md:88]: Done

File: documentation/Specification.md (hunk 9 above)
Comment (ljte) [reply documentation/Specification.md:115]: Done

File: documentation/Specification.md (hunk 11 above)
Comment (ljte) [reply documentation/Specification.md:118]: Done

**Review by inauros — APPROVED** [review]

**Pull request merged**